sdk/protocol-plugins/src/spark/SparkProtocolPlugin.ts

```diff
@@ -1,18 +1,234 @@
+  filterAssetsListByEMode,
+} from '../implementation/AAVEv3LikeBuilder'
+import { UNCAPPED_SUPPLY, PRECISION_BI } from '../implementation/constants'
+
+export class SparkProtocolPlugin extends BaseProtocolPlugin<SparkPoolId> {
+  public static protocol: ProtocolName.Spark = ProtocolName.Spark
+  public static supportedChains = valuesOfChainFamilyMap([ChainFamilyName.Ethereum])
+  public static schema = z.object({
+    protocol: z.object({
+      name: z.literal(ProtocolName.Spark),
+      chainInfo: z.object({
+        name: z.string(),
+        chainId: z.custom<ChainId>(
+          (chainId) =>
+            SparkProtocolPlugin.supportedChains.some((chainInfo) => chainInfo.chainId === chainId),
+          'Chain ID not supported',
+          true,
+        ),
+      }),
+    }),
+    emodeType: z.nativeEnum(EmodeType),
+  })
+
+  constructor() {
+    const StepBuildersMap = {
+      [SimulationSteps.DepositBorrow]: SparkDepositBorrowActionBuilder,
+    }
+    super(
+      SparkProtocolPlugin.protocol,
+      SparkProtocolPlugin.supportedChains,
+      SparkProtocolPlugin.schema,
+      StepBuildersMap,
+    )
   }
 
-  getActionBuilder<T extends steps.Steps>(step: T): Maybe<ActionBuilder<T>> {
-    return this.StepBuilders[step.type] as ActionBuilder<T>
+  async getPool(poolId: unknown): Promise<SparkLendingPool> {
+    this.isPoolId(poolId)
+    const emode = sparkEmodeCategoryMap[poolId.emodeType]
+
+    const ctx = this.ctx
+    const chainId = ctx.provider.chain?.id
+    if (!chainId) throw new Error('ctx.provider.chain.id undefined')
+
+    if (!SparkProtocolPlugin.supportedChains.some((chainInfo) => chainInfo.chainId === chainId)) {
+      throw new Error(`Chain ID ${chainId} is not supported`)
+    }
+
+    const builder = await new AaveV3LikePluginBuilder(ctx, SparkProtocolPlugin.protocol).init()
+    const reservesAssetsList = await builder
+      .addPrices()
+      .addReservesCaps()
+      .addReservesConfigData()
+      .addReservesData()
+      .addEmodeCategories()
+      .build()
+
+    const filteredAssetsList = filterAssetsListByEMode(reservesAssetsList, emode)
+
+    // Both USDC & DAI use fixed price oracles that keep both stable at 1 USD
+    const poolBaseCurrencyToken = CurrencySymbol.USD
+
+    const collaterals: Record<AddressValue, SparkPoolCollateralConfig> = {}
+    for (const asset of filteredAssetsList) {
+      const {
+        token: collateralToken,
+        config: { usageAsCollateralEnabled, ltv, liquidationThreshold, liquidationBonus },
+        caps: { supplyCap },
+        data: { totalAToken },
+      } = asset
+      const LTV_TO_PERCENTAGE_DIVISOR = 100n
+
+      try {
+        collaterals[collateralToken.address.value] = {
+          token: collateralToken,
+          price: Price.createFrom({
+            baseToken: collateralToken,
+            quoteToken: poolBaseCurrencyToken,
+            value: asset.price.toString(),
+          }),
+          priceUSD: Price.createFrom({
```

Comment:
Is this correct? price and priceUSD have the same value, but different quoteToken, which seems suspicious

Reply:
Oh, I see that poolBaseCurrencyToken is also USD